nexus/src/db/datastore.rs

```diff
@@ -1750,16 +1750,16 @@ impl DataStore {
         &self,
         opctx: &OpContext,
         authz_instance: &authz::Instance,
-        pagparams: &DataPageParams<'_, Name>,
     ) -> ListResultVec<NetworkInterface> {
         opctx.authorize(authz::Action::ListChildren, authz_instance).await?;
-
         use db::schema::network_interface::dsl;
-        paginated(dsl::network_interface, dsl::name, &pagparams)
+
+        dsl::network_interface
             .filter(dsl::time_deleted.is_null())
-            .filter(dsl::instance_id.eq(authz_instance.id()))
+            .filter(dsl::vpc_id.eq(authz_instance.id()))
```

Comment:
It seems you're checking if the `vpc_id` equals the instance ID — that can't be right.

Reply:
Thank you for your contribution. I have been meaning to make a similar change at some point.

Reply:
@david-crespo What's a good way to test this?

Reply:
I believe this would be caught by the existing integration tests. Specifically, [this call here](https://github.com/oxidecomputer/omicron/blob/c458fddcb8660eeb78d4a0ffe34c9d21c2ebd3dc/nexus/tests/integration_tests/instances.rs#L194) fetches the existing NICs attached to an instance. I can't see how that would return anything if the wrong UUID were used.

Can you show the output of your tests against this commit?

Reply:
If you look at CI on this commit, the tests passed even when that bug was in place

https://github.com/oxidecomputer/omicron/pull/1164/commits/c563bfe509f7b2105ae33641aaddf042fa8f7110